react/week2/homework/src/TodoList.js

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+import Counter from './Counter'
+import TodoItem from './TodoItem'
+import getTodos from './getTodos'
+
+function TodoList() {
+    const [todos, updateTodos] = useState(getTodos());
+
+    function addRandomTodo(todos, updateTodos) {
+        const newTodos = todos.concat({ id: generateUniqeId(), description: 'random todo' });
+        updateTodos(newTodos);
+    }
+
+    function handleRemove(id) {
+        const newTodos = todos.filter((aTodo) => aTodo.id !== id);
+        updateTodos(newTodos);
+    }
+
+    function generateUniqeId() {
+        return Math.random().toString(36).substr(2, 9);
+    }
```

Comment:
A comment about the `key` attribute:

For preference purposes, keys are used in React to identify which items in the list are changed. it helps to determine which item in a list needs to be rerendered instead of rerendering the entire set of list every time.

it's not recommended to assign `keys` as random values because each time the component rerenders it's value changes.


[https://reactjs.org/docs/reconciliation.html#tradeoffs](url)

> Keys should be stable, predictable, and unique. Unstable keys (like those produced by Math.random()) will cause many component instances and DOM nodes to be unnecessarily recreated, which can cause performance degradation and lost state in child components.


This is of course not in the scope of this homework but it's worth mentioning :)

Reply:
THank you for this remark! Really important!